util/src/main/java/org/killbill/billing/util/collect/Iterators.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020-2022 Equinix, Inc
+ * Copyright 2014-2022 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.util.collect;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.killbill.billing.util.Preconditions;
+
+/**
+ * Contains subset of Guava's {@link com.google.common.collect.Iterators} functionality.
```

Comment:
Could you add some language in the `util/README.md` that some of the code was taken out of `Guava version [XXX]` and add a link to the Guava repo on GitHub (+ mentions it's licensed using Apache License 2.0).

This is just to make sure we fulfill the redistributions / attributions requirements from the license.

Reply:
Done.